std/typecons.d

```diff
@@ -7143,3 +7143,265 @@ unittest
         ubyte, ubyte, T3, T3,
     );
 }
+
+/**
+ * Get the value of a field or property member. The member's name
+ * is resolved at runtime, rather than at compile-time.
+ *
+ * Note that currently only `pure @safe` property methods are supported.
```

Comment:
> Why the restriction to `pure` and `@safe`, then?

Two reasons:
- `pure @safe` is the minimum restriction required to filter out the problematic stuff from `Object`: `toString()` and `toHash()`.
- `nothrow` and `@nogc` are interesting mainly for performance reasons (whether latency or throughput). Such code shouldn't be using inefficient string-based runtime dispatch, anyway.

The garbage collector is an integral part of the D language; in the absence of a real alternative it shouldn't be banned without a good reason.